Looms.PoS/Looms.PoS.Persistance/Repositories/BusinessesRepository.cs

```diff
@@ -23,17 +23,31 @@ public async Task<BusinessDao> CreateAsync(BusinessDao businessDao)
 
     public async Task<IEnumerable<BusinessDao>> GetAllAsync()
     {
-        return await _context.Businesses.ToListAsync();
+        return await _context.Businesses.Where(x => !x.IsDeleted).ToListAsync();
     }
 
     public async Task<BusinessDao> GetAsync(Guid id)
     {
-        return await _context.Businesses.FindAsync(id)
-            ?? throw new LoomsNotFoundException("Business not found");
+        var businessDao = await _context.Businesses.FindAsync(id);
+
+        if (businessDao is null || businessDao.IsDeleted)
+        {
+            throw new LoomsNotFoundException("Business not found");
+        }
+        return businessDao;
+    }
+
+    public async Task<BusinessDao> UpdateAsync(BusinessDao businessDao)
+    {
+        await RemoveAsync(businessDao.Id);
```

Comment:
Why do you need to remove it first?

Reply:
Because we are using immutable records so we are creating and not updating. Need to remove the original to replace it with a new one. Otherwise primary key issue